kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/NetworkAdminServiceImpl.java

```diff
@@ -1714,10 +1644,168 @@ private int frequencyMhz2Channel(int frequency) {
+    }
+
+    private WifiMode getWifiMode(String ifaceName) throws KuraException {
+        WifiMode wifiMode = WifiMode.UNKNOWN;
+        List<? extends NetInterfaceConfig<? extends NetInterfaceAddressConfig>> netInterfaceConfigs = getNetworkInterfaceConfigs();
+        for (NetInterfaceConfig<? extends NetInterfaceAddressConfig> netInterfaceConfig : netInterfaceConfigs) {
+            if (netInterfaceConfig.getName().equals(ifaceName)) {
+                List<? extends NetInterfaceAddressConfig> netInterfaceAddresses = netInterfaceConfig
+                        .getNetInterfaceAddresses();
+                if (netInterfaceAddresses != null) {
+                    wifiMode = getWifiMode(netInterfaceAddresses);
+                }
+                break;
+            }
+        }
+        return wifiMode;
+    }
+
+    private WifiMode getWifiMode(List<? extends NetInterfaceAddressConfig> netInterfaceAddresses) {
+        WifiMode wifiMode = WifiMode.UNKNOWN;
+        for (NetInterfaceAddressConfig netInterfaceAddress : netInterfaceAddresses) {
+            if (netInterfaceAddress instanceof WifiInterfaceAddressConfig) {
+                wifiMode = ((WifiInterfaceAddressConfig) netInterfaceAddress).getMode();
+            }
+        }
+        return wifiMode;
+    }
+
+    private WifiSecurity getWifiSecurity(WifiAccessPoint wap) {
+        WifiSecurity wifiSecurity = WifiSecurity.NONE;
+
+        EnumSet<WifiSecurity> esWpaSecurity = wap.getWpaSecurity();
+        if (esWpaSecurity != null && !esWpaSecurity.isEmpty()) {
+            wifiSecurity = WifiSecurity.SECURITY_WPA;
+
+            Iterator<WifiSecurity> itWpaSecurity = esWpaSecurity.iterator();
+            while (itWpaSecurity.hasNext()) {
+                s_logger.trace("getWifiHotspots() :: WPA Security={}", itWpaSecurity.next());
+            }
+        }
+
+        EnumSet<WifiSecurity> esRsnSecurity = wap.getRsnSecurity();
+        if (esRsnSecurity != null && !esRsnSecurity.isEmpty()) {
+            if (wifiSecurity == WifiSecurity.SECURITY_WPA) {
+                wifiSecurity = WifiSecurity.SECURITY_WPA_WPA2;
+            } else {
+                wifiSecurity = WifiSecurity.SECURITY_WPA2;
+            }
+            Iterator<WifiSecurity> itRsnSecurity = esRsnSecurity.iterator();
+            while (itRsnSecurity.hasNext()) {
+                s_logger.trace("getWifiHotspots() :: RSN Security={}", itRsnSecurity.next());
+            }
+        }
+
+        if (wifiSecurity == WifiSecurity.NONE) {
+            List<String> capabilities = wap.getCapabilities();
+            if (capabilities != null && !capabilities.isEmpty() && capabilities.contains("Privacy")) {
+                wifiSecurity = WifiSecurity.SECURITY_WEP;
+            }
+        }
+
+        return wifiSecurity;
+
+    }
+
+    private void setCiphers(WifiHotspotInfo wifiHotspotInfo, WifiAccessPoint wap, WifiSecurity wifiSecurity) {
+        EnumSet<WifiSecurity> esWpaSecurity = wap.getWpaSecurity();
```

Comment:
maybe you could try to simplify the things by merging the sets together and having a single iterator. Furthermore, the checks seem to be repetitive, so maybe they could be extracted.